Loadable.svelte

```diff
@@ -40,11 +40,13 @@
     })
   }
 
-  export async function load(loader) {
+  export async function load(loader, unloader) {
     const componentModule = await loader()
     const component = componentModule.default || componentModule
 
-    LOADED.set(loader, component)
+    if (!unloader){
+      LOADED.set(loader, component)
+    }
     return component
   }
```

Comment:
I'm not sure what's happening here. If an `unloader` method is defined, why shouldn't it cache the loaded component?

Reply:
Before, you were *always* caching the component, and then, in the unMount handler (sorry I'm new to Svelte not sure if the callback you *return from* onMount has a better name) you were removing it. But this is a race condition. Imagine if the Loadable component loads, the component loading begins, and then the Loadable unMounts *before* the loading is finished.

In this case, the code to remove the (non-existent) component will run on unMount, do nothing, and then, when the async loading completes, the component will be cached, incorrectly, even if the user provided a truthy value for unloader. This change fixes that by simply refusing to cache it in the first place, if unloader is truthy, to side-step this race condition.

Reply:
Got it, but, as you said, we would be completely skipping the caching step even if the component didn't unmount. It could be loaded somewhere else and it wouldn't use the cache in this case.

Reply:
Not sure I follow. None of this is theoretic. I only discovered all of these issues when testing svelte-loadable. I purposefully slowed down my loading promises to see if the correct things happened. This PR represents fixes to what I discovered. 

In this case, the caching step was not ever skipped, even if it should have been. Before, the component was ***always*** cached, and then, if necessary, removed in unMount. This was a race condition because the unMount could run before the component was finished loading, and cached. The fix here is to just not cache it if it shouldn't be cached, rather than always cache it, and conditionally remove it from cache in unMount.